content/concepts/transports/webrtc.md

```diff
@@ -101,17 +101,81 @@ Authenticity is achieved by succeeding the
+Thus, this WebRTC private node to private node connectivity enables the vast majority of private browser nodes to connect with one another.
+{{< /alert >}}
+
+#### Transport Internals
+
+The libp2p WebRTC private-to-private transport is enabled by supporting the [W3C defined](https://w3c.github.io/webrtc-pc/#introduction) `RTCPeerConnection` API.
+This core API enables p2p connectivity and provides methods for establishing connections and transferring streams of data between peers.
+Running instances of libp2p that support this transport will have `/webrtc-private-to-private` in their multiaddr.
+
+However, there's more to p2p connections than what `RTCPeerConnection` provides. Crucially, signaling isn't built into the WebRTC API.
+{{< alert icon="" context="info">}}
+Signaling is the process of coordinating communication and exchanging metadata about the communication (i.e. initializing, closing, or reporting errors about connections).
+{{< /alert >}}
+
+For this transport, libp2p supports its own signaling protocol which has the protocol id: `webrtc-signaling`.
+
+#### How private-to-private connectivity works
 
 <!-- TO ADD DIAGRAM -->
 
+Suppose we have three network entities:
+
+- _Node A_ - a libp2p node running in the browser
+- _Node B_ - another browser libp2p node running on a different browser instance
+- _Relay R_ - a libp2p relay node
+
+In this connectivity scenario, _A_ wants to connect to _B_.
+This works as follows:
+
+- _B_ connects to _R_ and makes a relay reservation. It appends `webrtc-private-to-private` to its relayed multiaddress and advertises it to the network.
+- _A_ discovers _B_'s relayed multiaddr, sees it supports private-to-private, and establishes a relayed connection to _B_ via _R_.
+- _A_ and _B_ create outbound and inbound [`RTCPeerConnection` objects](https://webrtc.org/getting-started/peer-connections) respectively
+  - Per their namesake, these objects handle peer connections, define how they're set up, and contain other useful information about the network.
+- _A_ initiates the `webrtc-signaling` protocol to _B_ via a stream over the relayed connection
+  - This signaling stream is used to:
+    - Exchange SDPs between nodes (an offer from _A_ and an answer from _B_)
+    - Exchange information about the network connection (in WebRTC parlance this is called [exchanging ICE candidates](https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API/Connectivity#ice_candidates))
+- After this whole process is done, there can be one of two results:
+  - A successful direct connection is established between _A_ and _B_
+    - In this case, both browser nodes will close the signaling protocol stream
+    - The relayed connection is closed
+  - A failed direction connection
+    - In this case, the signaling stream is reset
+    - It's important to note that libp2p does not specify transferring data over the relayed connection if the direct connection fails
```

Comment:
Is this really necessary for newcomers to know?

Reply:
Removed.